src/test/java/com/fasterxml/jackson/databind/tofix/JsonMergeWithCustomCollection4783Test.java

```diff
@@ -0,0 +1,128 @@
+package com.fasterxml.jackson.databind.tofix;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.annotation.JsonMerge;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+// [databind#4783] Test to verify that JsonMerge also works for custom list
+public class JsonMergeWithCustomCollection4783Test
+{
+
+    private static class MyArrayListJDK<T> extends ArrayList<T> implements List<T> { }
+
+    static class MergeListJDK {
+        @JsonMerge
+        @JsonProperty
+        public List<String> values = create();
+        { values.add("a");}
+    }
+
+    static <T> List<T> create() {
+        return new MyArrayListJDK<T>();
+    }
+
+    interface MyListCustom<T> extends List<T> { }
+
+    private static class MyArrayListCustom<T> extends ArrayList<T> implements MyListCustom<T> { }
+
+    static class MergeCustomStringList {
+        @JsonMerge
+        @JsonProperty
+        public MyListCustom<String> values = createCustomStringList();
+        { values.add("a"); }
+    }
+
+    static <T> MyListCustom<T> createCustomStringList() {
```

Comment:
Not really needed as per above (but if it was, should parameterize with `String` to align with name.